lib/annotations.go

```diff
@@ -0,0 +1,35 @@
+package metadata
+
+const (
+	// CheckpointAnnotationContainerManager is used when creating an OCI image
+	// from a checkpoint archive to specify the name of container manager.
+	CheckpointAnnotationContainerManager = "org.criu.checkpoint.container.manager"
+
+	// CheckpointAnnotationName is used when creating an OCI image
+	// from a checkpoint archive to specify the name of the checkpoint.
+	CheckpointAnnotationName = "org.criu.checkpoint.container"
+
+	// CheckpointAnnotationPod is used when creating an OCI image
+	// from a checkpoint archive to specify the name of the pod associated with the checkpoint.
+	CheckpointAnnotationPod = "org.criu.checkpoint.pod"
+
+	// CheckpointAnnotationNamespace is used when creating an OCI image
+	// from a checkpoint archive to specify the namespace of the pod associated with the checkpoint.
+	CheckpointAnnotationNamespace = "org.criu.checkpoint.namespace"
```

Comment:
`"org.criu.checkpoint.pod.namespace"`

Reply:
`pod.namespace` seems strange to me. Because the pod is in a namespace. Should we do it hierarchically?

- namespace.name
- namespace.pod.name
- namespace.pod.uid
- namespace.pod.id
- namespace.pod.container.id
- namespace.pod.container.name

But that seems to complicated.

Reply:
According to the [rules for annotations](https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules), it is considered best practice to "namespace the keys":

> - Annotations MUST be a key-value map where both the key and value MUST be strings.
> - While the value MUST be present, it MAY be an empty string.
> - Keys MUST be unique within this map, and .
> - Keys SHOULD be named using a reverse domain notation - e.g. `com.example.myKey`.
> - The prefix `org.opencontainers` is reserved for keys defined in Open Container Initiative (OCI) specifications and MUST NOT be used by other specifications and extensions.
> - Keys using the `org.opencontainers.image` namespace are reserved for use in the OCI Image Specification and MUST NOT be used by other specifications and extensions, including other OCI specifications.
> - If there are no annotations then this property MUST either be absent or be an empty map.
> - Consumers MUST NOT generate an error if they encounter an unknown annotation key.